### DISCUSSION TITLE
QR timecode format on the HERO 12 Black

### DISCUSSION THREAD

**Discussion by PRS112233** [gopro/labs#640] [General]
Does anyone know if the format of the QR timecode on the HERO 12 Black is different from the format of the older models using GoPro Labs firmware?
On the UltraSync Blue slate iOS app when displaying QR timecode the HERO 12 says 'Invalid QR code' whereas on all other GoPro cameras running GoPro Labs the QR timecode is accepted and totally fine?

**Comment by dnewman-gpsw** [comment]
It is more strict.  HERO12 requires the formatting like here https://gopro.github.io/labs/control/precisiontime/  It needs time, daylight savings, zone info : oT231108082611.785oTD0oTZ-8

Add Labs to HERO12 as timecode is more flexible.

**Reply by PRS112233** [reply]
Thank you for the fast reply! I'll advise our customer to install GoPro Labs firmware on the HERO 12 Black rather than factory release firmware.

**Comment by williamkraushaar** [comment]
hey there, stumbled upon this discussion, with the same problem but I don't understand @dnewman-gpsw 's solution...my GoPro 12 won't read any QC code I create, I always get the "invalid QR code" message. everything worked fine with my GoPro 11, but this new 12 doesn't seem to accept any Qr code...

**Reply by dnewman-gpsw** [reply]
You likely don't have Labs installed, that message is from the stock camera firmware.  Look in the version in the camera menu under preferences->About->Camera Info to see the version number, all Labs version will end with .7x not .00.

**Reply by williamkraushaar** [reply]
oooh gotcha...of course! thanks!

**Reply by williamkraushaar** [reply]
btw, is there a way to "see" inside the camera if the added effects that you don't normally see as options (noise reduction, WIDE, bitrate, etc) appear or have been successfully added? not always easy to notice just by looking at the screen when filming. with my GoPro 11, all added labs effects would pop-up as the camera turned on, but with the 12 it seems like it just says "GoPro labs firmware etc"

**Reply by dnewman-gpsw** [reply]
On boot it lists the active features.  You can also scan oMFEAT=4 to display active features for 4 seconds.

**Reply by williamkraushaar** [reply]
interesting. only one of the effects (WIDE) seems to have been added successfully. the other few (BITR, NR01, LOGB) are not named. is this common?
thanks btw, really appreciating the help here

**Reply by dnewman-gpsw** [reply]
It means you haven't successful set BITR, NR01, LOGB

Try again:
![image](https://github.com/gopro/labs/assets/10856084/d23596aa-1699-4835-a3df-0e7479409d1e)

**Reply by williamkraushaar** [reply]
amazing. the other way it worked was by applying and scanning those effects one at a time. thanks a bunch!!!
